app/controllers/peer_reviews_controller.rb

```diff
@@ -1,67 +1,167 @@
+    @assignment = Assignment.find(params[:assignment_id])  # Needed for the following functions.
+    reviewer_groups = get_groupings_table_info()
+    reviewee_groups = get_groupings_table_info(@assignment.parent_assignment)
+    reviewee_to_reviewers_map = create_map_reviewee_to_reviewers(reviewer_groups, reviewee_groups)
+    id_to_group_names_map = create_map_group_id_to_name(reviewer_groups, reviewee_groups)
+    num_reviews_map = create_map_number_of_reviews_for_reviewer(reviewee_to_reviewers_map)
+    render json: [reviewer_groups, reviewee_groups, reviewee_to_reviewers_map,
+                  id_to_group_names_map, num_reviews_map]
   end
 
-  # GET /peer_reviews/new
-  def new
-    @peer_review = PeerReview.new
+  # Returns a dict of: reviewee_id => [list of reviewers].
+  def create_map_reviewee_to_reviewers(reviewer_groups, reviewee_groups)
+    reviewer_ids = []
+    reviewer_groups.each { |reviewer| reviewer_ids.push(reviewer['id']) }
+
+    peer_review_map = {}
+    reviewee_groups.each { |reviewee| peer_review_map[reviewee['id']] = [] }
+
+    # Populate a dictionary where each key is the reviewee, and the value is
+    # a list of all the reviewers. This may leave some entries empty, which
+    # means they have no reviewers.
+    peer_reviews = PeerReview.where(reviewer_id: reviewer_ids)
+    peer_reviews.each do |peer_review|
+      reviewee_group_id = peer_review.result.submission.grouping.id
+      peer_review_map[reviewee_group_id].push(peer_review.reviewer.id)
+    end
+
+    return peer_review_map
   end
 
-  # GET /peer_reviews/1/edit
-  def edit
+  # Returns a map of group id => names.
+  def create_map_group_id_to_name(reviewer_groups, reviewee_groups)
+    # We need to get every possible group so we have a big map of everyone that
+    # is present in both tables. This means ids from both the reviewers and the
+    # reviewees group, since this data is eligible for use in both tables.
+    unique_group_ids = {}
+    reviewer_groups.each { |reviewer| unique_group_ids[reviewer['id']] = 0 }
+    reviewee_groups.each { |reviewee| unique_group_ids[reviewee['id']] = 0 }
+
+    # Compress into a single list so we can pass it off as a query.
+    id_to_group_name_list = []
+    unique_group_ids.each do |key, val|
+      id_to_group_name_list.push(key)
+    end
+
+    # Retrieve all the groups with the unique id list, and map the id => name.
+    id_to_group_name_map = {}
+    groupings = Grouping.where(id: id_to_group_name_list)
+    groupings.each do |grouping|
+      id_to_group_name_map[grouping.id] = grouping.group.group_name
+    end
+
+    return id_to_group_name_map
   end
 
-  # POST /peer_reviews
-  # POST /peer_reviews.json
-  def create
-    @peer_review = PeerReview.new(peer_review_params)
+  # TODO - should this go in the view logic instead of here? (may needed for random assign though)
+  # Returns a map of reviewer_id => num_of_reviews
+  def create_map_number_of_reviews_for_reviewer(reviewee_to_reviewers_map)
```

Comment:
This whole method strikes me as a little odd. Why aren't we simply accessing peer reviews from the database and doing a `count` method?